src/care/array_utils.h

```diff
@@ -186,6 +186,15 @@ CARE_HOST_DEVICE bool checkSorted(const care::host_device_ptr<const T>& array, c
                                   const char* name, const char* argname,
                                   const bool allowDuplicates = false);
 
+#if defined(CARE_ENABLE_IMPLICIT_CONVERSIONS)
+
+template <typename T>
```

Comment:
Does this need to be #if defined(CARE_ENABLE_IMPLICIT_CONVERSIONS) protected? 

Could we just have both 
bool checkSorted(const care::host_device_ptr<T>& array
and also  
bool checkSorted(const care::host_device_ptr<const T>& array 
, without any #if defined(IMPLICIT) ?

Reply:
This function is only necessary if CARE_ENABLE_IMPLICIT_CONVERSIONS=ON, so I think it best to avoid code bloat and also unnecessary layers of function calls since they just make debugging harder.

Reply:
If that is the case, what do you think of other functions with the same paradigm? For example, right below checkSorted we have a similar paradigm for BinarySearch. Should you have the same #if statements there, or not or save that for a different pull request?

template<typename mapType>
CARE_HOST_DEVICE int BinarySearch(const care::host_device_ptr<const mapType>& map, const int start,
                                  const int mapSize, const mapType num,
                                  bool returnUpperBound = false) ;
template<typename mapType>
CARE_HOST_DEVICE int BinarySearch(const care::host_device_ptr<mapType>& map, const int start,
                                  const int mapSize, const mapType num,
                                  bool returnUpperBound = false) ;

Reply:
I've added an issue for fixing those. I think that should go in a separate pull request.

Reply:
Ok, no problem.